src/util/demangle.cpp

```diff
@@ -18,16 +18,21 @@
 
 #include "demangle.hpp"
 
+#include <regex>
+
 // Windows symbol demangler
 #ifdef _WIN32
-    // Turn off clang-format to avoid moving platform.h after Dbghelp.h
-    // (Dbghelp.h depends on types from Windows.h)
-    // clang-format off
-#    include "platform.hpp"
-#    include <Dbghelp.h>
 
-#    include <mutex>
-// clang-format on
+    // Dbghelp.h depends on types from Windows.h so it needs to be included first
+    // the define keeps clang-tidy from moving it
+    #ifdef _WIN32
+        #include "platform.hpp"
+    #endif
+
+    #include <Dbghelp.h>
```

Comment:
I would think this should be in the macro guard too

Reply:
The second macroguard is to stop clang format sorting. If you put the debug in there as a system include it gets sorted above platform. Then everything breaks

Reply:
You may as well just keep the clang-format off/on then. Having the second macro guard is just weird and confusing

Reply:
I changed the separator to a comment